src/blt_mad/helpers.go

```diff
@@ -171,20 +171,25 @@ func GetValuesLargerThanPercentile(numbers []float64, percentile float64) []floa
 	return largerValues
 }
 
-func SaveArrayToFile(fileName string, arr []float64) error {
-	file, err := os.Create(fileName)
+func AppendFloat64ArrayToTxt(fileName string, floatArray []float64) error {
```

Comment:
open-ended question: rather than manually write individual values into a file, could you reduce this to a couple of json marshall calls? this has the advantage that you can rely on the libraries to read/write and probably reduce your code to a few lines depending on what you drop into the file. (It might imply specifying yet another data structure with json metadata.)